### DISCUSSION TITLE
unable to download videos

### DISCUSSION THREAD

**Discussion by xena2025** [aclap-dev/video-downloadhelper#1471] [Help and Q&A]
### Your question

unable to download videos HLS from 

https://www.astrokaksha.in/s/courses/658dc834e4b03c13b16f1c9b/take
![image](https://github.com/user-attachments/assets/8c215919-eebb-474b-b5bd-a122a225b409)

<html>
<body>

Tab | 788347568 (incognito: true)
-- | --
BASIC TO ADVANCE VEDIC JYOTISH - Course
https://www.astrokaksha.in/s/courses/658dc834e4b03c13b16f1c9b/take with 12 headers
https://www.astrokaksha.in/s/videos/65b61414e4b065c2aeee6639/cover?v=1
true
https://qcdn.spayee.in/spees/w/o/658bf54ee4b02674c4a9b24e/v/65b61414e4b065c2aeee6639/u/673707dc112b773ff437723c/t/279de9d4d637bc91ef883b03ddd43922/p/assets/videos/658bf54ee4b02674c4a9b24e/2024/01/28/65b61414e4b065c2aeee6639/hls_1M_.m3u8
https://qcdn.spayee.in/spees/w/o/658bf54ee4b02674c4a9b24e/v/65b61414e4b065c2aeee6639/u/673707dc112b773ff437723c/t/279de9d4d637bc91ef883b03ddd43922/p/assets/videos/658bf54ee4b02674c4a9b24e/2024/01/28/65b61414e4b065c2aeee6639/hls_1M_.m3u8
none
Mp4 b:RawHls p:hls 7748.799999999493s
H264 at -1, fps: ?, q:?, no dim

</body>
</html>

### Addon & CoApp version and details

version: 9.1.1.0
target: google
channel: stable
lang: en-US
coapp: {"found":true,"new_version":false,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19"}
license: {"accepted":true,"email":"<removed>","key":"2efeceb7975341f8"}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/130.0.0.0 Safari/537.36
{
    "valid_license_message_has_been_displayed": true,
    "used_history_button": true,
    "last_download_directory": {
        "__serializer_tag": "option_some",
        "__serializer_value": {
            "__serializer_tag": "primitive",
            "__serializer_value": "L:\\Mittal"
        }
    },
    "view_options": {
        "all_tabs": false,
        "hide_downloaded": false,
        "low_quality": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": true,
        "sort_by_status": true,
        "sort_reverse": false
    },
    "open_count_store": 23,
    "successfull_dl": 89,
    "never_show_successfull_dl_message": true
}

**Comment by Mattmjs** [comment]
ffmpeg says _Unable to open key file_ https://qcdn.spayee.in/spees/w/o/658bf54ee4b02674c4a9b24e/v/65b61414e4b065c2aeee6639/u/673707dc112b773ff437723c/t/279de9d4d637bc91ef883b03ddd43922/p/assets/videos/658bf54ee4b02674c4a9b24e/2024/01/28/65b61414e4b065c2aeee6639/k/timestamp

Inside the playlist file I find this:

#EXT-X-KEY:METHOD=AES-128,URI="k/timestamp",IV=0x496daa1c6914000e408c65cead91fc29

I think your out of luck here, looks like the video uses encryption to prevent downloading.
